### ISSUE TITLE
(UWP) .NET Native Compiler crashes or mrt100_app.dll Runtime crashes with NETCore 5.3.3

### ISSUE THREAD

**Issue by WilliamABradley** [dotnet/roslyn#20773]
**Versions Used**: 
VS Community 2017 15.2 and 15.3 Preview
Microsoft.NETCore.UniversalWindowsPlatform 5.3.3
Microsoft.NETCore.UniversalWindowsPlatform 5.4.0
UWP Target Version: 10.0.14393.0
UWP Min Version: 10.0.14393.0
(Using PackageReferences in .csproj files)
FFmpeg and FFmpegInterop (Precompiled, I'm reasonably sure this isn't causing an issue, but is worth noting).

**Steps to Reproduce**:

1. Initially I had Copy if newer set on a couple of Resource Dictionaries.
This seemed to have caused a Null Reference Exception for the .NET Native compiler in VS Community 2017, however, in 15.3, An error said that it failed Generating Resources. (This might also have been caused by NuGet.Build.Packaging). *I can't seem to Replicate this issue anymore for some reason.
2. Removing this seemed to have fixed that issue.
I found there was an update to Microsoft.NETCore.UniversalWindowsPlatform 5.4.0, which might be why it compiled also. (This update is no longer listed).

I'm assuming that because 5.4.0 contains .NET Native 1.7 and an updated Compiler, that is why it compiles, however, I also assume that the Windows Dev Center won't accept that version yet (I'm looking to Publish ASAP).
I have compared my project files to my last known compile-able version, and it doesn't appear to be that different, so I have no idea why I suddenly struggle with compiling.

*Last time I ran into an issue like this (with my previous app) in #18737, it was due to a bug with System.Linq and Tuples. That took a long time to discover.

**Expected Behavior**:
Successful Compile

**Actual Behavior**:
If using UWP 5.3.3:
C:\Users\Willi\.nuget\packages\microsoft.net.native.compiler\1.6.2\tools\Microsoft.NetNative.targets(691,5): error : Internal compiler error: Object reference not set to an instance of an object.

If upgraded to UWP 5.4.0:
Compiles and runs correctly, however, unable to submit to the Windows Store as .NET Native 1.6 and 1.7 were above the minimum available versions.

**In my Test App, If upgraded to UWP 5.4.0, then downgraded to 5.3.3**:
It compiles, but I get an immediate runtime error:
`Unhandled exception at 0x0FEADB2B (mrt100_app.dll) in Lib-Test.exe: 0xC0000602:  A fail fast exception occurred. Exception handlers will not be invoked and the process will be terminated immediately.`
@ 
`global::System.Threading.Volatile.Write(ref s_initialized, 2);` in Helpers.g.cs.

Switching my Test App back to UWP 5.4.0:
Compiles and runs correctly.

I looked at my .resw files and my Resource Dictionaries (as mrt is Modern Resource Technology), they seemed fine, no invalid characters in names.

I can send someone a copy of the private Repo of my App if that helps.
I guess what I need to know is, Will 5.4.0 be available soon? Or is there a temporary fix?

**Comment by jcouv** [comment]
@MichalStrehovsky This looks like a crash in the .NET Native toolset. Please advise.

**Comment by MichalStrehovsky** [comment]
Sorry for the trouble!

I would suggest reaching out to the dotnetnative at Microsoft.com email address listed here: https://github.com/dotnet/core/blob/master/Documentation/ilcRepro.md

A full stack of the FailFast would help with initial triage, but Helpers.g.cs sounds like a problem in interop. We're better equipped to help out over email.

**Comment by WilliamABradley** [comment]
Do you guys know when 5.4.0 will be available for use? Fall creators update?

I'll email them. Thanks.

**Comment by WilliamABradley** [comment]
@MichalStrehovsky I have sent the ILC Repo with the Subject **(UWP) .NET Native Compiler crashes with NETCore 5.3.3**.

When should I hear back about this? I'm looking to deploy the app as soon as possible.

Alternatively, do you know when Microsoft.NETCore.UniversalWindowsPlatform 5.4.0 will be usable in the Store?

**Comment by WilliamABradley** [comment]
I'm closing this because my App now compiles after #21206, however, this occurred before I Converted my Resources to ResX in a .NET Standard Library.

**Comment by WilliamABradley** [comment]
Turns out this issue still exists, and is a separate issue from #21206, I had my project still set to Microsoft.NETCore.UniversalWindowsPlatform 5.4.0, which is why it compiled again.
